### DISCUSSION TITLE
Brew search not finding formulae/casks that I can see online

### DISCUSSION THREAD

**Discussion by adamcc** [Homebrew/discussions#3718] [Everyday usage]
Does anyone understand this?
```
% brew search miniconda
==> Formulae
minica                                minicom                               minidlna
```
But https://formulae.brew.sh/cask/miniconda#default

and 
```
% brew search mambaforge
Error: No formulae or casks found for "mambaforge".
```
But https://formulae.brew.sh/cask/mambaforge

Is there something wrong with the installation?

**Comment by SMillerDev** [comment]
I'm gonna guess you're on linux where casks don't exist. Note how none of the first ones say "miniconda" and the header says Formulae

**Reply by adamcc** [reply]
I'm on a Mac, I can see some casks.
```
% brew search linear
==> Formulae
liblinear

==> Casks
homebrew/cask-fonts/font-noto-sans-linear-a                              homebrew/cask-fonts/font-noto-sans-linear-b
```

**Reply by gromgit** [reply]
Is `homebrew/cask` tapped? If:
```
ls -l $(brew --repo homebrew/cask)
```
gets you an error, you need to:
```
brew tap homebrew/cask
```

**Reply by adamcc** [reply]
@SMillerDev Thanks, you're onto something. I'm on a Mac, and it appears searching formulae works consistently.

However, I can see a subset of casks:
```
% brew search linear
==> Formulae
liblinear

==> Casks
homebrew/cask-fonts/font-noto-sans-linear-a                              homebrew/cask-fonts/font-noto-sans-linear-b
```
But it's true that many or most are not found by brew search. 

For example, if I just search for things listed on the [list of casks](https://formulae.brew.sh/cask/), few are found, and sometimes casks are found that aren't listed on the webpage.

```
% brew search 1password       
==> Casks
homebrew/cask-versions/1password-beta           homebrew/cask-versions/1password-cli-beta       homebrew/cask-versions/1password-cli1
```

But the webpage lists 
- 1password 
- 1password-cli

**Reply by adamcc** [reply]
@gromgit 
Thanks, that seems to have worked.

```
% brew search mamba
==> Formulae
samba  mame

==> Casks
mambaforge     mame     mamp      micromamba
```

Needing to do that seems to conflist with the [Cask page](https://github.com/Homebrew/homebrew-cask) that says "To start using [Homebrew](https://brew.sh/) Cask, you just need Homebrew installed."

**Reply by SMillerDev** [reply]
The distinction is that if you want to install something you don't need to tap anything (because brew will). If you want to search it can't possibly know you mean to search cask files too without tapping first.

**Reply by adamcc** [reply]
I don't follow. It found some casks. The logic is mysterious to me. Anyhow, thanks for your help.

From the [docs](https://docs.brew.sh/Manpage#search-texttext):
"The search for `text` is extended online to `homebrew/core` and `homebrew/cask`."

**Reply by apurv-jay** [reply]
> I'm gonna guess you're on linux where casks don't exist. Note how none of the first ones say "miniconda" and the header says Formulae

I am on linux trying to use brew but i cannot see packages in brew
warning: No available formula with the name "sublime-text".
==> Searching for similarly named formulae...
Error: No formulae found for sublime-text.

**Reply by gromgit** [reply]
`sublime-text` is a cask (binary package provided by the developer), which is not supported on Linux. Since upstream isn't releasing source code, your only option is to [use the appropriate package repo for your system](https://www.sublimetext.com/docs/linux_repositories.html).
